modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -682,15 +682,23 @@ class Session implements ISession {
         if( !aborted ) {
             joinAllOperators()
             log.trace "Session > all operators finished"
+            final finalizerComplete = finalizePoolManager?.shutdown(false)
+            final publisherComplete = publishPoolManager?.shutdown(false)
+            if( !finalizerComplete || !publisherComplete ) {
+                final ignoreErrors = config.navigate('workflow.output.ignoreErrors')
+                if( !ignoreErrors )
+                    throw new AbortOperationException("Timed out while waiting to publish outputs")
+            }
+        }
+        else {
+            finalizePoolManager?.shutdown(true)
+            publishPoolManager?.shutdown(true)
```

Comment:
Moving the shutdown here is effectively the same as before because `destroy()` is called right after `await()` in the script runner.

Although, `destroy()` is not called in the event of an abort, which means that previously the thread pools were only ever shut down when `aborted == false`.

So we can either keep these lines here, move them to `abort()`, or remove them entirely, since they were previously unreachable anyway